sharedbuffers/mapped_struct.py

```diff
@@ -494,46 +786,32 @@ def __delitem__(self, index):
         raise AttributeError("Proxy objects are read-only")
 
     def __iter__(self):
-        for i in xrange(self.objlen):
+        for i in xrange(len(self)):
             yield self[i]
```

Comment:
You probably want a `_get` method or something that you can call in an optimized fashion instead of this generic getitem. Iteration I imagine will be a big use case for proxied containers, so you probably want them to be efficient.